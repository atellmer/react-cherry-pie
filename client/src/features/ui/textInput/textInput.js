/** @flow */
import React from 'react';
import { HOC as withFormsy } from 'formsy-react';
import TextField from 'material-ui/TextField/index';


type Props = {
  name: string,
  title: string,
  type: string,
  getValue: Function,
  setValue: Function,
  getErrorMessage: Function
};

const TextInput = (props: Props) => {
  const {
    name,
    title,
    type,
    getValue,
    setValue,
    getErrorMessage
  } = props;

  const changeValue = (e: Event & {target: {value: string}}) => {
    setValue(e.target.value);
  };

  return (
    <TextField
      name={name}
      type={type || 'text'}
      hintText={title || ''}
      fullWidth
      onChange={changeValue}
      value={getValue()}
      errorText={getErrorMessage()} />
  );
};

export { TextInput };
export default withFormsy(TextInput);
