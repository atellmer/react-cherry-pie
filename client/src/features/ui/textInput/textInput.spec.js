import React from 'react';
import { shallow } from 'enzyme';

import { TextInput } from './textInput';


describe('TextInput component', () => {
  let props;

  beforeEach(() => {
    props = {
      name: 'login',
      title: 'login',
      type: 'text',
      getValue: jest.fn(),
      setValue: jest.fn(),
      getErrorMessage: jest.fn()
    };
  });

  test('should match to snapshot', () => {
    const component = shallow(<TextInput {...props} />);

    expect(component).toMatchSnapshot();
  });

  test('should call setValue with argument', () => {
    const component = shallow(<TextInput {...props} />);

    component.find('TextField').simulate('change', { target: { value: 'Hello world' } });
    expect(component.instance().props.setValue).toHaveBeenCalledWith('Hello world');
  });

  test('should change type correctly', () => {
    const component = shallow(<TextInput {...props} />);

    component.setProps({ type: 'email' });

    expect(component.find('TextField').node.props.type).toBe('email');
  });
});
