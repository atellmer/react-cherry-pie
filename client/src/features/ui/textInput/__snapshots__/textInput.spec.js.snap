// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component should match to snapshot 1`] = `
<TextField
  disabled={false}
  floatingLabelFixed={false}
  fullWidth={true}
  hintText="login"
  multiLine={false}
  name="login"
  onChange={[Function]}
  rows={1}
  type="text"
  underlineShow={true}
/>
`;
